File: library/src/main/java/me/proxer/library/api/info/TranslatorGroupEndpoint.java
package me.proxer.library.api.info;

import lombok.experimental.Accessors;
import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;
import me.proxer.library.entitiy.info.TranslatorGroup;

/**
 * Endpoint for retrieving all information of an {@link TranslatorGroup}.
 *
 * @author Ruben Gees
 */
@Accessors(fluent = true)
public class TranslatorGroupEndpoint implements Endpoint<TranslatorGroup> {

    private final InternalApi internalApi;

    private final String id;

    TranslatorGroupEndpoint(final InternalApi internalApi, final String id) {
        this.internalApi = internalApi;
        this.id = id;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public ProxerCall<TranslatorGroup> build() {
        return internalApi.translatorGroup(id);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/info/TranslatorGroupEndpoint.kt
package me.proxer.library.api.info

import me.proxer.library.ProxerCall
import me.proxer.library.api.Endpoint
import me.proxer.library.entity.info.TranslatorGroup

/**
 * Endpoint for retrieving all information of an [TranslatorGroup].
 *
 * @author Ruben Gees
 */
class TranslatorGroupEndpoint internal constructor(
    private val internalApi: InternalApi,
    private val id: String
) : Endpoint<TranslatorGroup> {

    override fun build(): ProxerCall<TranslatorGroup> {
        return internalApi.translatorGroup(id)
    }
}
